packages/astro/src/runtime/server/render/component.ts

```diff
@@ -164,13 +164,12 @@ async function renderFrameworkComponent(
 	} else {
 		// Attempt: use explicitly passed renderer name
 		if (metadata.hydrateArgs) {
-			const passedName = metadata.hydrateArgs;
-			const rendererName = rendererAliases.has(passedName)
-				? rendererAliases.get(passedName)
-				: passedName;
-			renderer = renderers.find(
-				({ name }) => name === `@astrojs/${rendererName}` || name === rendererName
-			);
+			const rendererName = metadata.hydrateArgs;
+			if (clientOnlyValues.has(rendererName)) {
+				renderer = renderers.find(
+					({ name }) => name === `@astrojs/${rendererName}` || name === rendererName
+				);
+			}
```

Comment:
I haven't run this, but I think the changes here will prevent `client:only="solid"` from having an effect anymore.

We will try to find a renderer by the name of `@astrojs/solid` but the package name is `@astrojs/solid-js`.

Is it possible to reduce the changes here? I'm asking because it will also make reviewing easier since we could be rest assured that behavior doesn't change - only the messaging.

Reply:
Thanks. I'm keeping the same logic as before.